internal/metrics/metrics_test.go

```diff
@@ -162,10 +163,27 @@ func (cl *client) Delete(name string) error {
 }
 
 func checkMetrics(c *qt.C, url, substr string, expectedLines []string) {
-	resp, err := http.DefaultClient.Get(url)
-	c.Assert(err, qt.Equals, nil)
-	defer resp.Body.Close()
-	lines := linesContaining(c, resp.Body, substr)
+	timeout := time.After(5 * time.Second)
+	tick := time.Tick(100 * time.Millisecond)
+	lines := make([]string, 0, len(expectedLines))
+	getLines := func() []string {
+		resp, err := http.DefaultClient.Get(url)
+		c.Assert(err, qt.Equals, nil)
+		defer resp.Body.Close()
+		return linesContaining(c, resp.Body, substr)
+	}
+loop:
+	for {
+		select {
+		case <-timeout:
+			break loop
+		case <-tick:
```

Comment:
So, am I understanding that both timeout and tick are separate process, and this for loop is catching either of them firing? So that each time tick goes off we run getlines and when timeout goes off we stop?